pkg/internal/server/grpc/get_metadata_delta_test.go

```diff
@@ -703,3 +705,89 @@ func validateCSIGetMetadataDeltaRequest(t *testing.T, csiReq, expectedCSIReq *cs
 	assert.Equal(t, csiReq.MaxResults, expectedCSIReq.MaxResults)
 	assert.Equal(t, csiReq.Secrets, expectedCSIReq.Secrets)
 }
+
+func TestStreamGetMetadataDeltaResponse(t *testing.T) {
+	ctx := context.Background()
+	th := newTestHarness().WithMockCSIDriver(t).WithFakeClientAPIs()
+	defer th.TerminateMockCSIDriver()
+
+	grpcServer := th.StartGRPCServer(t, th.Runtime())
+	defer th.StopGRPCServer(t)
+
+	for _, tc := range []struct {
+		name              string
+		setCSIDriverReady bool
+		req               *api.GetMetadataDeltaRequest
+		mockCSIResponse   bool
+		mockCSIError      error
+		expectStreamError bool
+		expStatusCode     codes.Code
+		expStatusMsg      string
+	}{
+		{
+			name:              "success",
+			setCSIDriverReady: true,
+			req: &api.GetMetadataDeltaRequest{
+				BaseSnapshotName:   "snap-1",
+				TargetSnapshotName: "snap-2",
+			},
+			mockCSIResponse:   true,
+			mockCSIError:      nil,
+			expectStreamError: false,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+
+			// TODO: Refactor whole test execution to mock client based on test case
+			grpcServer.CSIDriverIsReady()
+			mockController := gomock.NewController(t)
```

Comment:
The `runtime.TestHarness` defines a mock controller: `MockController` which we can expose in this test harness from `WithMockCSIDriver(t)` if desired.  If this is done, though, one will have to create an instance of the test harness each iteration to ensure that the `Finish` method is invoked in the correct context.